client/src/components/App/Content/SignUp/SignUp.tsx

```diff
@@ -27,6 +27,8 @@ const SignUp : React.FC<Props> = ({ open, handleSaveUser, handleCloseSignUp }) =
 
     const userId = useSelector<StoreStateType, string>(state => state.user.id);
 
+    const [isFormValid, setIsFormValid] = React.useState<boolean>(false);
```

Comment:
I think you can use isValid api from react-hook-form instead of implement it yourself. I don't know if it will work but worth a try